lib/services/task.ts

```diff
@@ -247,4 +278,79 @@ export class TaskService {
       return { success: false, error: 'Failed to update task order' };
     }
   }
+  // Sync tasks
+  // Pulls pending changes from Dexie and sends them to the server
+  // Marks Dexie tasks synced after successful sync
+  // returns the count of synced tasks
+  async syncTasks(userId: string): Promise<ServiceResponse<number>> {
```

Comment:
Are we using the userId here? I don't think so, and that would mean we're not filtering out other user todos stored in indexdb.